framework/include/materials/MaterialData.h

```diff
@@ -191,6 +191,11 @@ class MaterialData
    */
   bool isOnlyResizeIfSmaller() const { return _resize_only_if_smaller; };
 
+  /**
+   * Remove the property storage and element pointer from MaterialPropertyStorage data structures
+   */
+  void eraseProperty(const Elem * elem) { _storage.eraseProperty(elem); };
```

Comment:
Maybe add a sentence to the doc comment saying "Use this when elements are deleted so we don't end up with invalid elem pointers (for e.g. stateful properties) hanging around in our data structures." - or something like that.

Reply:
I will add what you suggest. Thanks.